### PR TITLE
Item category list

### PR DESCRIPTION
**Pull request by AxioBen**
Fix https://github.com/worldskills/worldskills-infrastructure-list/issues/7

### CODE REVIEW

**Comment by AxioBen** [comment]
@fabien This one can be checked 👍

File: app/scripts/controllers/itemcategory.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * # ItemCategoryCtrl
+ * Controller of the ilApp
+ */
+angular.module('ilApp')
+  .controller('ItemCategoryCtrl', function ($q, $scope, $state, $confirm, $translate, $aside, ItemCategory, WSAlert, APP_ROLES) {
+
+    $scope.event = false;
+    $scope.data = {};
+    $scope.APP_ROLES = APP_ROLES;
+    $scope.loadingEvent = $q.defer();
```
Comment (fabian) [inline app/scripts/controllers/itemcategory.js]: What's the purpose of `$scope.loadingEvent`?

File: app/scripts/app.js [hunk 2]
```diff
@@ -278,6 +278,21 @@ angular
...
+    url: '/event/{eventId}/item-category',
+    templateUrl: 'views/item-category.html',
+    controller: 'ItemCategoryCtrl',
+    data: {
+      requireLoggedIn: true,
+      requiredRoles: [
+        {code: 2200, role: APP_ROLES.ADMIN },
+        {code: 2200, role: APP_ROLES.ORGANIZER },
+        {code: 2200, role: APP_ROLES.WS_MANAGER },
+        {code: 2200, role: APP_ROLES.WS_SECTOR_MANAGER }
```
Comment (fabian) [inline app/scripts/app.js]: Not all theses roles should have access, only "Admin" and "EditItemCategories" should have access.

File: app/styles/application.scss [hunk 3]
```diff
@@ -389,3 +389,46 @@ legend.aside{ padding: 0; font-size: 100%; margin-bottom: 1em; }
 .b{
   font-weight:bold;
 }
+
+$brand-primary: #003d56;
```
Comment (fabian) [inline app/styles/application.scss]: Is this used anywhere? Please remove if not.

File: app/views/event.overview.html [hunk 4]
```diff
@@ -69,6 +69,7 @@ <h5 class="list-group-item-heading">{{skill.name.text}} <span class="badge small
       <div class='panel-heading' translate>ADMIN_VIEWS</div>
       <div class='panel-body'>
         <ul>
+          <li><a ui-sref="itemCategory({eventId: selectedEvent.id})" translate>ITEM_CATEGORY_AND_SUBCATEGORY</a></li>
```
Comment (fabian) [inline app/views/event.overview.html]: Can you check if the logged in user has the role `Admin` or `EditItemCategories` and show the link only for those users?

File: app/views/item-category.html [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+          <tr ng-repeat="item in data.subCategories">
+              <td>{{item.name.text}}</td>
+              <td>{{item.description.text}}</td>
+              <td>{{item.parent.name.text}}</td>
+              <td class="actions">
+                  <a class="btn btn-danger btn-xs"  ng-click="removeItemCategory(item, false)"><span class="glyphicon glyphicon-remove"></span></a>
+                  <a class="btn btn-warning btn-xs" ng-click="openItemEditor(item, false, $index)"><span class="glyphicon glyphicon-edit"></span></a>
+              </td>
+          </tr>
+        </tbody>
```
Comment (fabian) [inline app/views/item-category.html]: Can you add a fallback text "No subcategories found." if there are no sub categories?

File: app/views/item-category.html [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        <tbody>
+          <tr ng-repeat="item in data.categories">
+              <td>{{item.name.text}}</td>
+              <td>{{item.description.text}}</td>
+              <td class="actions">
+                  <a class="btn btn-danger btn-xs" ng-click="removeItemCategory(item, true)"><span class="glyphicon glyphicon-remove"></span></a>
+                  <a class="btn btn-warning btn-xs" ng-click="openItemEditor(item, true, $index)"><span class="glyphicon glyphicon-edit"></span></a>
+              </td>
+          </tr>
+        </tbody>
```
Comment (fabian) [inline app/views/item-category.html]: Also add a fallback text "No categories found." if there are no categories here.

File: app/views/item-category.html [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      <div>
+        <div class="pull-left">
+            <button class="btn btn-sm btn-success" ng-click="openItemEditor({}, true)">
+              <span class="glyphicon glyphicon-plus"></span>&nbsp;<span translate>ADD_CATEGORY</span>
+            </button>
+        </div>
+        <div class="clearfix"></div>
+      </div>
+    </div>
+    <div class="table-responsive mt-3">
```
Comment (fabian) [inline app/views/item-category.html]: What's the meaning of `mt-3`?

File: app/views/edititemcategoryaside.html [hunk 8]
```diff
@@ -0,0 +1,43 @@
+<div class="aside ng-scope">
```
Comment (fabian) [inline app/views/edititemcategoryaside.html]: Can you rename this file to `item-category-aside.html`? I realize there's already `editsupplieditemaside.html`, but that's a really bad example. 😉

File: app/views/item-category.html (hunk 7 above)
Comment (AxioBen) [reply app/views/item-category.html]: That's a helper class used by our integrator. I add mt-3 class in the application.scss now.

**Comment by AxioBen** [comment]
@fabian You can re-check this PR 👍

**Pull request merged**